app/admin/rate_management/pricelist_items.rb

```diff
@@ -0,0 +1,190 @@
+# frozen_string_literal: true
+
+ActiveAdmin.register RateManagement::PricelistItem, as: 'Pricelist Item' do
+  menu false
+  actions :index
+  decorate_with RateManagementPricelistItemDecorator
+  belongs_to :rate_management_pricelist, parent_class: RateManagement::Pricelist
+
+  controller do
+    def scoped_collection
+      super.preload(
+        :routing_group,
+        :vendor,
+        :account,
+        :routeset_discriminator,
+        :gateway,
+        :gateway_group,
+        :routing_tag_mode,
+        dialpeer: %i[
+          gateway
+          gateway_group
+          routing_tag_mode
+        ]
+      )
+    end
+
+    # @see InheritedResources::BelongsToHelpers#parent
+    def parent
+      return @parent if defined?(@parent)
+
+      parent_model = association_chain[-1]
+      @parent = parent_model ? RateManagementPricelistDecorator.decorate(parent_model) : nil
+    end
+  end
+
+  filter :dialpeer_id_eq, label: 'Dialpeer ID'
+  filter :prefix
+  acts_as_filter_by_routing_tag_ids routing_tag_ids_count: true
+  filter :initial_rate
+  filter :next_rate
+  filter :connect_fee
+  filter :initial_interval
+  filter :next_interval
+  filter :dst_number_min_length
+  filter :dst_number_max_length
+  boolean_filter :enabled
+  filter :priority
+  filter :valid_from
+  boolean_filter :valid_from_null, label: 'Valid From NOW'
+
+  sidebar 'Rate Management Pricelist', only: :index, priority: 0 do
+    attributes_table_for(helpers.parent) do
+      row :id
+      row :name
+      row :project
+      row :state, &:state_badge
+      row :background_job, &:background_job_badge
+      row :dialpeers, &:dialpeers_link
+      row :retain_enabled
+      row :retain_priority
+      row :valid_till
+      row :applied_at if helpers.parent.applied?
+      row :created_at
+    end
+  end
+
+  action_item :detect_dialpeers do
+    if authorized?(:detect_dialpeers) && helpers.parent.new? && !helpers.parent.has_background_job?
+      link_to 'Detect Dialpeers',
+              detect_dialpeers_rate_management_pricelist_path(helpers.parent)
+    end
+  end
+
+  action_item :redetect_dialpeers do
+    if authorized?(:redetect_dialpeers) && helpers.parent.dialpeers_detected? && !helpers.parent.has_background_job?
+      link_to 'Redetect Dialpeers',
+              redetect_dialpeers_rate_management_pricelist_path(helpers.parent)
+    end
+  end
+
+  action_item :apply_changes do
+    if authorized?(:apply_changes) && helpers.parent.dialpeers_detected? && !helpers.parent.has_background_job? && !helpers.parent.items.with_error.exists?
```

Comment:
on index add before_action

```ruby
detect_dialpeers_in_progress? || model.apply_changes_in_progress?

if helpers.parent.dialpeers_detected? && helpers.parent.detect_dialpeers_in_progress?
  flash[:warning] = 'Redetect Dialpeers background job is in progress'
elsif helpers.parent.detect_dialpeers_in_progress?
  flash[:warning] = 'Detect Dialpeers background job is in progress'
elsif helpers.parent.apply_changes_in_progress?
  flash[:warning] = 'Apply Changes background job is in progress'
elsif helpers.parent.items.with_error.exists?
  flash[:warning] = 'Changes can't be applied because pricelist has error items'
end
```